tests/reftests/filter-variable-resolution.test

```diff
@@ -0,0 +1,1087 @@
+ "foo" { ?v:smtg }
+ "undefined-smtg" { !?v:smtg }
+]
+### opam install depends --show
+The following actions would be performed:
+=== install 2 packages
+  - install depends        1
+  - install undefined-smtg 1 [required by depends]
+### :III:7: depopts :::
+### <pkg:depopts.1>
+opam-version: "2.0"
+depopts: [
+ "bar" { ?v:smtg }
+ "undefined-smtg" { !?v:smtg }
+]
+### opam install depopts
+The following actions will be performed:
+=== install 1 package
+  - install depopts 1
+
+<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
+-> installed depopts.1
+Done.
+### opam install bar --show
+The following actions would be performed:
+=== install 1 package
+  - install bar 1
+### opam install undefined-smtg --show
+The following actions would be performed:
+=== recompile 1 package
+  - recompile depopts        1 [uses undefined-smtg]
+=== install 1 package
+  - install   undefined-smtg 1
+### opam remove depopts
+The following actions will be performed:
+=== remove 1 package
+  - remove depopts 1
+
+<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
+-> removed   depopts.1
+Done.
+### :III:8: conflicts :::
+### <pkg:conflicts.1>
+opam-version: "2.0"
+conflicts: [
+  "baz" { ?v:smtg }
+  "undefined-smtg" { !?v:smtg }
+]
+### opam install conflicts
+The following actions will be performed:
+=== install 1 package
+  - install conflicts 1
+
+<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
+-> installed conflicts.1
+Done.
+### opam install baz --show
+The following actions would be performed:
+=== install 1 package
+  - install baz 1
+### opam install undefined-smtg --show
+The following actions would be performed:
+=== remove 1 package
+  - remove  conflicts      1 [conflicts with undefined-smtg]
+=== install 1 package
+  - install undefined-smtg 1
+### :III:9: msg :::
+### <pkg:msg.1>
+opam-version: "2.0"
+messages: [
+  "Hej!" { ?v:smtg }
+  ">>smtg is undefined<<" { !?v:smtg }
+]
+### opam install msg
+The following actions will be performed:
+=== install 1 package
+  - install msg 1  Hej!
+
+<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
+-> installed msg.1
+Done.
+### :III:10: post-msg :::
+### <pkg:post-msg.1>
+opam-version: "2.0"
+post-messages: [
+  "Bye!" { ?v:smtg }
+  ">>smtg is undefined<<" { !?v:smtg }
+]
+### opam install post-msg
+The following actions will be performed:
+=== install 1 package
+  - install post-msg 1
+
+<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
+-> installed post-msg.1
+Done.
+
+<><> post-msg.1 installed successfully ><><><><><><><><><><><><><><><><><><><><>
+=> Bye!
+### :III:11: avail :::
+### <pkg:avail.1>
+opam-version: "2.0"
+available: [ ?v:smtg ]
+### opam install avail
+[ERROR] avail: unmet availability conditions: '?v:smtg'
+# Return code 5 #
+### :::::::::::::::::::::::::::
+### :: Self package variable ::
+### :::::::::::::::::::::::::::
```

Comment:
for this section i think it would be also be useful to show the behaviour of opam when the package has already been installed. Like, for each fields, show the behaviour as is done currently, then force-install the package and try to install it again and show if there is a difference of behaviour

Reply:
updated with these tests in a separate commit